### DISCUSSION TITLE
IIdael Plugin host (v22.10) return infos

### DISCUSSION THREAD

**Discussion by subderground** [DISTRHO/Cardinal#406] [Q&A]
Hello,
Everything is still great for me :-) but here are some infos about the IIdeal Plugin host on v22.10.
- Open Internal plugin -> everything OK (no custom/original gui)
- Open VST3 (64bit) -> sound OK but no sliders/parameters and no custom/original gui plugin available. Depend on VST3, if the sliders/parameters appears, the Reaper crashes when I try to move one.
- Open JSFX -> sound and sliders/parameters OK but no custom/original gui available.
- Open LV2 -> everything ok but I only tested with your "ProM" plugin :-)

Just for info (but it's not a real problem); when inside the lldael Plugin host, the spacebar does not 'play/stop' the DAW anymore. It works after a simple click outside the menu of this module. 

It would be so powerful to have in the future a cv in&out on each slider/parameter... If possible ;-)

Ps: I'am on Win11 64bit Reaper (No Carla installed)

**Comment by falkTX** [comment]
VST3 hosting in Ildaeil is incomplete, I am surprised you even are able to load anything with it.
It is not even 50% done from Carla side.
A possible solution would be to enable the juce stuff on Cardinal's Carla build, but really do not want to make things more bloated than what they already are.
CLAP hosting takes priority now, though that already seems to be working well.

JSFX is normal to not have "GUI"s, those are not implemented yet.

Automating parameters is in progress, there is a WIP PR for it, but needs to be redone.

So basically:
- stick with CLAP and VST2 if loading external plugins
- do not use VST3 hosting
- JSFX do not have gui, as expected
- CLAP hosting to be finalized first, then auto-scanning, then param automation

**Reply by subderground** [reply]
No problem, I'm just curious so I have tested it. :-)
For the JSFX gui, sorry I was thinking about visual spectrum analyser or stuff like picture below (picture attached)... but that's really not important (sliders are fine for me in the modular world).  
![gui-jsfx](https://user-images.githubusercontent.com/105162700/196052819-3fb7fad9-4587-4067-8546-463e2b72006d.jpg)

**Reply by falkTX** [reply]
yeah I know of those. the ysfx project that provides these UIs only has macos, windows and generic graphics via JUCE, but as I stated before I want to avoid that as dependency.
and not a priority, I am also fine with just sliders.
